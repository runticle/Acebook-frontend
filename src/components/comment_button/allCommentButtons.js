import React from 'react';
import EditButton from './editButton';
import NewCommentButton from './newCommentButton';
import LikeButton from './likeButton';
import DeleteButton from './deleteButton';

class AllCommentButtons extends React.Component {

  render() {

    return (
              // ADD IN COMMENT ID WHEN INFO IS AVAILABLE FROM FETCH REQUEST
              <div id="buttons">
                < DeleteButton post_id ={ this.props.post_id } />
                < EditButton post_id ={ this.props.post_id } />
                < NewCommentButton post_id ={ this.props.post_id } />
                < LikeButton post_id ={ this.props.post_id } />
              </div>
          )
    }
}

export default AllCommentButtons;
