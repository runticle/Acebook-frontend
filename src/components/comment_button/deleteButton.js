import React from 'react';
import config from '../../config'
import { getAuthenticationHeaders } from '../../actions/auth'


export class DeleteButton extends React.Component {

  deletePostReq() {
    var id = this.props.id
    var post_id = this.props.post_id
    console.log(`${config.URL}/posts/${post_id}/comments/${id}`)
    fetch(`${config.URL}/posts/${post_id}/comments/${id}`, {
      method: 'delete',
      headers: getAuthenticationHeaders()
    })
      .then(res => res.json())
      .then(
        (result) => {
          console.log(result)
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          console.log(error)
          this.setState({
            error
          });
        }
      )
  }

  handleDelete = event => {
    event.preventDefault();
    this.deletePostReq();
  }

  render() {

    return (
      <div id="deleteButton">
        <button id="delete" onClick={ this.handleDelete } className="button">❌</button>
      </div>
          )
    }
}

export default DeleteButton;
