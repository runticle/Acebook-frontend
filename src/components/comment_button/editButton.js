import React from 'react';


class EditButton extends React.Component {
  //
  // editPostReq() {
  //   fetch("https://acebook-stars.herokuapp.com/posts", {
  //     method: 'patch',
  //     id: this.props.post_id
  //   })
  //     .then(res => res.json())
  //     .then(
  //       (result) => {
  //         console.log(result)
  //       },
  //       // Note: it's important to handle errors here
  //       // instead of a catch() block so that we don't swallow
  //       // exceptions from actual bugs in components.
  //       (error) => {
  //         console.log(error)
  //         this.setState({
  //           error
  //         });
  //       }
  //     )
  // }

  handleEdit = event => {
    event.preventDefault();
    console.log(this.props.post_id)
  }

  render() {

    return (
      <div id="editButton">
        <button id="edit" onClick={ this.handleEdit } className="button">✏️</button>
      </div>
          )
    }
}

export default EditButton;
