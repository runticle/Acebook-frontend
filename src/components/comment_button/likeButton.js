import React from 'react';


class LikeButton extends React.Component {
  //
  // likePostReq() {
  //   fetch("https://acebook-stars.herokuapp.com/posts", {
  //     method: 'post',
  //     id: this.props.post_id
  //   })
  //     .then(res => res.json())
  //     .then(
  //       (result) => {
  //         console.log(result)
  //       },
  //       // Note: it's important to handle errors here
  //       // instead of a catch() block so that we don't swallow
  //       // exceptions from actual bugs in components.
  //       (error) => {
  //         console.log(error)
  //         this.setState({
  //           error
  //         });
  //       }
  //     )
  // }

  handleLike = event => {
    event.preventDefault();
    console.log(this.props.post_id)
  }

  render() {

    return (
      <div id="likeButton">
        <button id="toggle_like" onClick={ this.handleLike } className="button">⭐️</button>
      </div>
          )
    }
}

export default LikeButton;
