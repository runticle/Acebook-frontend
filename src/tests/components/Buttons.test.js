import React from 'react'
import { shallow } from 'enzyme'
import { AllCommentButtons } from '../../components/comment_button/allCommentButtons'

let wrapper = shallow(<AllCommentButtons />)

test('should render Buttons correctly', () => {
  expect(wrapper).toMatchSnapshot();
})
