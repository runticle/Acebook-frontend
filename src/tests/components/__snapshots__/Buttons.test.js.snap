// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Buttons correctly 1`] = `
<div
  id="buttons"
>
  <DeleteButton />
  <EditButton />
  <NewCommentButton />
  <LikeButton />
</div>
`;
